cmd/mixtool/server.go

```diff
@@ -99,58 +98,47 @@ type ruleProvisioner struct {
 // to existing, does not provision them. It returns whether Prometheus should
 // be reloaded and if an error has occurred.
 func (p *ruleProvisioner) provision(r io.Reader) (bool, error) {
-	b := bytes.NewBuffer(nil)
-	tr := io.TeeReader(r, b)
-
-	f, err := os.Open(p.ruleFile)
-	if err != nil && !os.IsNotExist(err) {
-		return false, fmt.Errorf("open rule file: %w", err)
-	}
-	if os.IsNotExist(err) {
-		f, err = os.Create(p.ruleFile)
-		if err != nil {
-			return false, fmt.Errorf("create rule file: %w", err)
-		}
+	fileData, err := ioutil.ReadFile(p.ruleFile)
+	if err != nil {
+		return false, fmt.Errorf("unable to read %w", err)
```

Comment:
By convention, and to improve legibility of wrapped errors, we normally put a colon `:` between our error and the error being wrapped.

Reply:
ok!